### ISSUE TITLE
bug(styles): focus outline 

### ISSUE THREAD

**Issue by platon-rov** [SAP/fundamental-styles#3541]
### Describe the bug

Here is two big issues about the focus outline.

1. Focus outline cannot be rounded in Safari. It doesn't work like that.

<img width="49" alt="image" src="https://user-images.githubusercontent.com/20265336/172190312-acc82e6e-cb24-4ba5-891b-7419066505b8.png">

2. In Fiori, Fiori HC, Horizon HC (7 of 9 themes in total) we should have **rectangular** focus outline. In Chromium-based & Firefox focus outline inherits border-radius from the element. Only in Horizon Evening & Morning focus should be **rounded**.

Fiori
<img width="908" alt="image" src="https://user-images.githubusercontent.com/20265336/172131451-fb3c8f21-f41f-40af-9207-f35d924db53a.png">

Horizon
<img width="259" alt="image" src="https://user-images.githubusercontent.com/20265336/172184519-9bf2c212-241e-4bf2-a3cd-463ea0f01fef.png">

Horizon HC
<img width="209" alt="image" src="https://user-images.githubusercontent.com/20265336/172131176-0902949f-8372-486a-883d-ff84a7663c35.png">

### Possible approach

Use pseudo elements (before/after) to create focus outlines that won't be relying on elements' border-radius and, also, these outlines may be rounded in Safari.

Pros: 
* Focus outline 100% per specs.
* Safari issue solved.

Cons:
* Somewhere both of the pseudo elements already in use.
* A lot of work to do (incl. specific spacings).

**Comment by platon-rov** [comment]
Not per specs (rounded outline but should be rectangular in fiori, horizon hc):
* Button
* Calendar
* Card
* Generic Tile
* Grid List
* Product Switch
* Token

Per specs but broken in safari (outline cannot be rounded in Horizon):
* Checkbox
* Input Group
* Input
* Radio
* Textarea
* Select
* Slider
* Step Input
* Switch

Not sure (no outline design specs found):
* Avatar?
* Avatar Group?
* Busy indicator?
* Carousel?
* Rating Indicator?
* Menu? First & Last
* Micro Process Flow?
* Object Marker?
* User Menu?
* Tree?
* List?
* Icon Tab Bar?
* Tabs?
* Status Indicator?
* Side Navigation?
* Vertical Navigation?
* Horizontal Navigation?
* Dynamic Page?

Under the "Not sure" section listed all the components for which outline specs not fully clear, for example Avatar - its outline should be rectangular in Fiori, Horizon HC but should it be rectangular/rounded/circle in Horizon? The same for rest. In case for certain component from the list outline should be rectangular in Horizon we don't need to take any actions against that component.

**Comment by IB002** [comment]
No chrome and Firefox issue is no longer relevant:

<img width="117" alt="Screenshot 2023-07-20 at 9 18 03 AM" src="https://github.com/SAP/fundamental-styles/assets/132930816/f6b36955-24b8-4e52-b814-3006262c37b6">
<img width="119" alt="Screenshot 2023-07-20 at 9 18 11 AM" src="https://github.com/SAP/fundamental-styles/assets/132930816/f6a0f9b9-7d68-4acf-9c87-28278c85195a">
<img width="124" alt="Screenshot 2023-07-20 at 9 18 26 AM" src="https://github.com/SAP/fundamental-styles/assets/132930816/fc29577e-ff73-47a1-a8b8-f345346ea228">

Safari is able to round buttons:

Input:
<img width="598" alt="Screenshot 2023-07-20 at 9 14 26 AM (2)" src="https://github.com/SAP/fundamental-styles/assets/132930816/658ed476-7701-40ba-929a-4608a16c47c9">

Slider
<img width="1190" alt="Screenshot 2023-07-20 at 9 15 40 AM (2)" src="https://github.com/SAP/fundamental-styles/assets/132930816/e65ceca6-56d5-4d59-a3bb-f0d5d730648c">
